File: group-anagrams/group-anagrams.py
class Solution:
    def key_by_sort(self, strs):
        return ''.join(sorted(strs))

    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        sortMap = {}
        for s in strs:
            key = self.key_by_sort(s)
            if key in sortMap:
                sortMap[key].append(s)
            else:
                sortMap[key] = [s]
        return list(sortMap.values())

File: group-anagrams/group-anagrams.js
//https://leetcode.com/problems/group-anagrams/

const keyBySortedChar = (s) => [...s].sort().join('');

var groupAnagrams = function(strs) {
    const sortMap = new Map();
    for (let i = 0; i < strs.length; i++) {
        const key = keyBySortedChar(strs[i]);
        if (sortMap.has(key)) {
            sortMap.get(key).push(strs[i])
        } else {
            sortMap.set(key, [strs[i]])
        }
    };
    return [...sortMap.values()]
};